src/machine.ts

```diff
@@ -78,6 +81,58 @@ export const syncHandler = () => (callback, onReceive) => {
   };
 };
 
+// @ts-ignore
+export const mempoolHandler = () => (callback, onReceive) => {
+  logger.debug('Mempool handler instantiated');
+  const iterator = syncLatestMempool();
+  const asyncCall: () => void = async () => {
+    for (;;) {
+      const txResult: GeneratorYieldResult<MempoolEvent> = await iterator.next();
+      const { value, done } = txResult;
+
+      if (done) {
+        // The generator reached its end, we should end this handler
+        logger.debug('Done.', value)
+        break;
+      }
+
+      if (value && !value.success) {
+        logger.error(value.message);
+        // if there is any error calling the lambda, wait 30 seconds and retry
+        if (value.type == 'wait') {
+          callback('WAIT');
+          return;
+        }
+        callback('ERROR');
```

Comment:
I like the retry mechanism! Maybe we should add that to the block sync as well 

For what types of errors will we transition to the `failure` state? Maybe add a comment explaining

Reply:
`failure` will be entered when the fullnode mempool api fails, any other error could be mitigated by waiting for the next block (which should take up to 30s).